w3-provider.md

```diff
@@ -13,463 +12,353 @@
+[Customer] with an active [subscription] MAY provision [space] with set of capabilities offered by the [subscription]. [Customer] MAY set limits on the space provision to manage costs, which makes provisioning [space] controlled by third party financially viable.
 
-## Creation
+[Subscription] can be conceptualized as a leasing agreement between [provider] and [customer], in which case [provision] can be viewed as subleasing agreement between [customer] and a [space].
 
-User MAY create a new space by generating a [keypair][public key cryptography] and deriving a valid [did:key][] identifier from it.
+### Customer
 
-> It is RECOMMENDED that user facing applications create a _space_ for a new user with a [ED25519][] keypair & delegate capabilities to it to a local agent whose DID is derived from another [non-extractable keypair](https://developer.mozilla.org/en-US/docs/Web/API/CryptoKey#cryptokey.extractable).
->
-> This grants an agent access to a space without reusing its key or a risk of it been compromised.
+Customer is a user [account] that has a [subscription] with a service [provider].
 
-```json
-// illustration of the space to agent delegation
-{
-  "iss": "did:key:zSpace",
-  "aud": "did:key:zAgent",
-  "exp": null, // delegation never expires
-  // allows did:key:zAgent to do anything with did:key:zSpace
-  "att": [
-    {
-      "with": "did:key:zSpace",
-      "can": "*"
-    }
-  ]
-}
-```
+ℹ️ Customer does not need to own or even have access to a [space], e.g. employer MAY setup an [account], setup [subscription] and use it to [provision] employee [space]s.
 
-## Setup
+### Space
 
-As we have established, space creator is an owner and has a full authority to delegate whichever capabilities to others. However, unless a space has an active provider of the capabilities, no invocation of them could succeed.
+Namespace, or space for short, is an owned resource that can be shared. It corresponds to the asymmetric keypair and is identified by the [`did:key`] URI. Space can be provisioned with capabilities provided by the [provider] through a [subscription].
 
-To make capabilities invocable, one needs to obtain a provider and add it to the desired space. For example, a user could get the "free" provider from web3.storage, which provides `store/*` and `upload/*` capabilities allowing them to store up to 5GiB of data.
+# Capabilities
 
-```mermaid
-flowchart TB
-  Space((did:key:zAliceSpace))
-  W3{{did:web:free.web3.storage}}
-  NFT{{did:web:nft.storage}}
+Here we describe protocol for arranging subscriptions and using it to provision [space]s.
 
-  Name([name/*])
-  Upload([upload/*])
-  Store([store/*])
+## Provider Capabilities
 
-  style W3 fill:grey,color:black,stroke:grey
-  style NFT fill:grey,color:black,stroke:grey
+### Provider Add
 
+User MAY invoke `provider/add` capability on the [account] subject (`with` field) to start a subscription with [provider] (`aud` field).
```

Comment:
```suggestion
A user MAY invoke the `provider/add` capability on the [account] subject (`with` field) to start a subscription with a [provider] (`aud` field).
```

Hm, this is a bit confusing - does "`with` field" refer to the invocation UCAN? Would be great to clarify this. Similar for "`aud` field" - I feel like that needs a bit more context.